app/src/main/java/com/kickstarter/ui/fragments/RewardsFragment.kt

```diff
@@ -48,25 +51,54 @@ class RewardsFragment : BaseFragment<RewardsFragmentViewModel.ViewModel>(), Rewa
         this.viewModel.outputs.showPledgeFragment()
                 .compose(bindToLifecycle())
                 .compose(observeForUI())
-                .subscribe { showPledgeFragment(it.first, it.second) }
+                .subscribe {
+                    dialog.dismiss()
+                    showPledgeFragment(it.first, it.second)
+                }
 
         this.viewModel.outputs.showAddOnsFragment()
                 .compose(bindToLifecycle())
                 .compose(observeForUI())
                 .subscribe {
+                    dialog.dismiss()
                     showAddonsFragment(it)
                 }
 
         this.viewModel.outputs.rewardsCount()
                 .compose(bindToLifecycle())
                 .compose(observeForUI())
                 .subscribe { setRewardsCount(it) }
+        
+        this.viewModel.outputs.showAlert()
+                .compose(bindToLifecycle())
+                .compose(observeForUI())
+                .subscribe {
+                    showAlert()
+                }
 
         context?.apply {
             ViewUtils.setGone(rewards_count, ViewUtils.isLandscape(this))
         }
     }
 
+    private fun createDialog() {
+        context?.let { context ->
+            dialog = AlertDialog.Builder(context, R.style.AlertDialog)
+                    .setCancelable(false)
+                    .setTitle(getString(R.string.Continue_with_this_reward))
+                    .setMessage(getString(R.string.It_may_not_offer_some_or_all_of_your_add_ons))
+                    .setNegativeButton("             ${getString(R.string.No_go_back)}") { _, _ -> {} }
+                    .setPositiveButton("             ${getString(R.string.Yes_continue)}") { _, _ ->
+                        this.viewModel.inputs.alertButtonPressed()
+                    }.create()
+        }
+    }
+
+    private fun showAlert() {
+        dialog.create()
```

Comment:
do we need to call `create()` here? We are also calling it in the `createDialog()` function, so it would be called twice.

Reply:
You are completely right! removing it 👍